ModelMapper mapping error hdbsequence

Having the following hdbsequence:
```
schema = "my_schema";
start_with = 1;
maxvalue = 10000000;
nomaxvalue = false;
minvalue= 1;
nominvalue = true;
cycles = false;
depends_on_table = "my_table";
reset_by = "SELECT IFNULL(MAX(\"Id\"), 0) + 1 FROM \"my_schema\".\"my_table\"";
```
After successful creation, the exception below is logged regularly
```
2021-11-10 09:48:01.440 [ERROR] [pool-5486-thread-1] o.e.d.c.s.api.AbstractSynchronizer - Resource [/registry/public/project_name/SEQUENCES/sequennce_nname.hdbsequence] skipped due to an error: org.modelmapper.MappingException: ModelMapper mapping errors:

1) Error mapping com.sap.xsk.parser.hdbsequence.models.XSKHDBSEQUENCEModel to com.sap.xsk.hdb.ds.model.hdbsequence.XSKDataStructureHDBSequenceModel

1 error
org.eclipse.dirigible.core.scheduler.api.SynchronizationException: org.modelmapper.MappingException: ModelMapper mapping errors:

1) Error mapping com.sap.xsk.parser.hdbsequence.models.XSKHDBSEQUENCEModel to com.sap.xsk.hdb.ds.model.hdbsequence.XSKDataStructureHDBSequenceModel

1 error
	at com.sap.xsk.hdb.ds.facade.XSKHDBCoreFacade.handleResourceSynchronization(XSKHDBCoreFacade.java:105) ~[xsk-modules-engines-hdb-0.11.0-SNAPSHOT.jar:na]
	at com.sap.xsk.hdb.ds.synchronizer.XSKDataStructuresSynchronizer.synchronizeResource(XSKDataStructuresSynchronizer.java:390) ~[xsk-modules-engines-hdb-0.11.0-SNAPSHOT.jar:na]
	at org.eclipse.dirigible.core.scheduler.api.AbstractSynchronizer.synchronizeCollection(AbstractSynchronizer.java:106) ~[dirigible-core-scheduler-6.1.3.jar:na]
	at org.eclipse.dirigible.core.scheduler.api.AbstractSynchronizer.synchronizeCollection(AbstractSynchronizer.java:113) ~[dirigible-core-scheduler-6.1.3.jar:na]
	at org.eclipse.dirigible.core.scheduler.api.AbstractSynchronizer.synchronizeCollection(AbstractSynchronizer.java:113) ~[dirigible-core-scheduler-6.1.3.jar:na]
	at org.eclipse.dirigible.core.scheduler.api.AbstractSynchronizer.synchronizeRegistry(AbstractSynchronizer.java:90) ~[dirigible-core-scheduler-6.1.3.jar:na]
	at com.sap.xsk.hdb.ds.synchronizer.XSKDataStructuresSynchronizer.synchronizeRegistry(XSKDataStructuresSynchronizer.java:375) ~[xsk-modules-engines-hdb-0.11.0-SNAPSHOT.jar:na]
	at com.sap.xsk.hdb.ds.synchronizer.XSKDataStructuresSynchronizer.synchronize(XSKDataStructuresSynchronizer.java:217) ~[xsk-modules-engines-hdb-0.11.0-SNAPSHOT.jar:na]
	at org.eclipse.dirigible.core.scheduler.api.AbstractSynchronizerJob$1.run(AbstractSynchronizerJob.java:52) ~[dirigible-core-scheduler-6.1.3.jar:na]
	at org.eclipse.dirigible.commons.timeout.TimeLimited$1.call(TimeLimited.java:38) ~[dirigible-commons-timeout-6.1.3.jar:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:829) ~[na:na]
Caused by: org.modelmapper.MappingException: ModelMapper mapping errors:

1) Error mapping com.sap.xsk.parser.hdbsequence.models.XSKHDBSEQUENCEModel to com.sap.xsk.hdb.ds.model.hdbsequence.XSKDataStructureHDBSequenceModel

1 error
	at org.modelmapper.internal.Errors.throwMappingExceptionIfErrorsExist(Errors.java:380) ~[modelmapper-2.4.0.jar:na]
	at org.modelmapper.internal.MappingEngineImpl.map(MappingEngineImpl.java:80) ~[modelmapper-2.4.0.jar:na]
	at org.modelmapper.ModelMapper.mapInternal(ModelMapper.java:573) ~[modelmapper-2.4.0.jar:na]
	at org.modelmapper.ModelMapper.map(ModelMapper.java:406) ~[modelmapper-2.4.0.jar:na]
	at com.sap.xsk.hdb.ds.parser.hdbsequence.XSKHDBSequenceParser.parseHanaXSClassicContent(XSKHDBSequenceParser.java:111) ~[xsk-modules-engines-hdb-0.11.0-SNAPSHOT.jar:na]
	at com.sap.xsk.hdb.ds.parser.hdbsequence.XSKHDBSequenceParser.parse(XSKHDBSequenceParser.java:61) ~[xsk-modules-engines-hdb-0.11.0-SNAPSHOT.jar:na]
	at com.sap.xsk.hdb.ds.service.parser.XSKCoreParserService.parseDataStructure(XSKCoreParserService.java:34) ~[xsk-modules-engines-hdb-0.11.0-SNAPSHOT.jar:na]
	at com.sap.xsk.hdb.ds.facade.XSKHDBCoreFacade.handleResourceSynchronization(XSKHDBCoreFacade.java:93) ~[xsk-modules-engines-hdb-0.11.0-SNAPSHOT.jar:na]
	... 13 common frames omitted
Caused by: java.lang.ClassCastException: null
```

Consider removing this modelmapper and using the antl4 parser to set the necessary fields

--- a/modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/model/hdbsequence/XSKDataStructureHDBSequenceModel.java
+++ b/modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/model/hdbsequence/XSKDataStructureHDBSequenceModel.java
@@ -12,6 +12,7 @@
 package com.sap.xsk.hdb.ds.model.hdbsequence;
 
 import com.sap.xsk.hdb.ds.model.XSKDataStructureModel;
+import com.sap.xsk.parser.hdbsequence.models.XSKHDBSEQUENCEModel;
 import com.sap.xsk.parser.hdbsequence.utils.HDBSequenceConstants;
 
 public class XSKDataStructureHDBSequenceModel extends XSKDataStructureModel {
@@ -27,6 +28,23 @@ public class XSKDataStructureHDBSequenceModel extends XSKDataStructureModel {
   private String dependsOnTable;
   private String dependsOnView;
   private Boolean publicProp = HDBSequenceConstants.PUBLIC_DEFAULT_VALUE;
+
+  public XSKDataStructureHDBSequenceModel(){}
+
+  public XSKDataStructureHDBSequenceModel(XSKHDBSEQUENCEModel model){
+    this.incrementBy = model.getIncrementBy();
+    this.startWith = model.getStartWith();
+    this.maxValue = model.getMaxValue();
+    this.noMaxValue = model.getNoMaxValue();
+    this.noMinValue = model.getNoMinValue();
+    this.minValue = model.getMinValue();
+    this.cycles = model.getCycles();
+    this.resetBy = model.getResetBy();
+    this.dependsOnTable = model.getDependsOnTable();
+    this.dependsOnView = model.getDependsOnView();
+    this.publicProp = model.getPublic();
+    this.setSchema(model.getSchema());
+  }
 
   public Integer getIncrementBy() {
     return incrementBy;

--- a/modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/parser/hdbsequence/XSKHDBSequenceParser.java
+++ b/modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/parser/hdbsequence/XSKHDBSequenceParser.java
@@ -39,9 +39,6 @@ import java.util.regex.Pattern;
 import org.antlr.v4.runtime.ANTLRInputStream;
 import org.antlr.v4.runtime.CommonTokenStream;
 import org.antlr.v4.runtime.tree.ParseTree;
-import org.modelmapper.ModelMapper;
-import org.modelmapper.config.Configuration;
-import org.modelmapper.convention.MatchingStrategies;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -102,13 +99,9 @@ public class XSKHDBSequenceParser implements XSKDataStructureParser {
         .registerTypeAdapter(XSKHDBSEQUENCEModel.class, new XSKHDBSEQUENCEModelAdapter())
         .create();
     XSKHDBSEQUENCEModel antlr4Model = gson.fromJson(parsedResult, XSKHDBSEQUENCEModel.class);
-    ModelMapper modelMapper = new ModelMapper();
-    modelMapper.getConfiguration()
-        .setFieldMatchingEnabled(true)
-        .setFieldAccessLevel(Configuration.AccessLevel.PRIVATE)
-        .setMatchingStrategy(MatchingStrategies.STRICT);
 
-    XSKDataStructureHDBSequenceModel hdbSequenceModel = modelMapper.map(antlr4Model, XSKDataStructureHDBSequenceModel.class);
+    XSKDataStructureHDBSequenceModel hdbSequenceModel = new XSKDataStructureHDBSequenceModel(antlr4Model);
+
     XSKHDBUtils.populateXSKDataStructureModel(location, content, hdbSequenceModel, IXSKDataStructureModel.TYPE_HDB_SEQUENCE,
         XSKDBContentType.XS_CLASSIC);
 

--- a/modules/parsers/parser-hdbsequence/src/main/java/com/sap/xsk/parser/hdbsequence/models/XSKHDBSEQUENCEModel.java
+++ b/modules/parsers/parser-hdbsequence/src/main/java/com/sap/xsk/parser/hdbsequence/models/XSKHDBSEQUENCEModel.java
@@ -153,16 +153,20 @@ public class XSKHDBSEQUENCEModel {
     this.dependsOnView = dependsOnView;
   }
 
+  public Boolean getPublic() { return this.publicProp; }
+
   public void setPublic(Boolean publicProp) {
     this.publicProp = publicProp;
   }
 
   @Override
   public boolean equals(Object o) {
-    if (this == o)
+    if (this == o) {
       return true;
-    if (o == null || getClass() != o.getClass())
+    }
+    if (o == null || getClass() != o.getClass()) {
       return false;
+    }
     XSKHDBSEQUENCEModel that = (XSKHDBSEQUENCEModel) o;
     return Objects.equals(missingProps, that.missingProps) && Objects.equals(schema, that.schema) && Objects.equals(incrementBy,
         that.incrementBy) && Objects.equals(startWith, that.startWith) && Objects.equals(maxValue, that.maxValue) && Objects.equals(

